enhancements/sig-architecture/30-clusterset-override/README.md

```diff
@@ -0,0 +1,358 @@
+# ManagedClusterSet Override
+
+## Release Signoff Checklist
+ 
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management/website/)
+ 
+## Summary
+The proposed work enhances the managedClusterSet API to support managedClusterSet Override.
+
+## Motivation
+Today, managedClusterSet is designed to group the managedClusters, a managedCluster can only be included in only one managedClusterSet, by setting a label `cluster.open-cluster-management.io/clusterset` on the managedCluster.
+A user must have explicit RBAC rules, `managedclustersets/join`, to add such labels.
+There are also raising requirements on using managedClusterSet for the following purposes. Within a purpose, the managedClusterSet must be exclusive, but for different purposes it can overlap.
+- As cluster admin, I want to create a managedClusterSet which includes all clusters in the apac region. Then I can apply certain configurations to all clusters in apac.
+- As a SRE team member, I want to use a managedClusterSet for Disaster Recovery purposes. The managedClusterSet should select two or more clusters in different regions. 
+- As a DEV team member, I want to use a managedClusterSet to select clusters based on the middleware enabled on these clusters, so I could run special applications in these clusters.
+
+## Proposal
+
+### Changes on managedClusterSet APIs.
+
+Currently, managedClusterSets use the label `cluster.open-cluster-management.io/clusterset:<ManagedClusterSet Name>` to select managedClusters. 
+But for some use cases, I want to use other labels to select managedClusters in a managedClusterSets.
+
+So, In this proposal, we change the managedClusterSets spec and want to provid a flexible way to select managedClusters. 
+
+```go
+type ManagedClusterSetSpec struct {
+    // Selector represents a selector of ManagedClusters by labels and names.
+    ClusterSelector ManagedClusterSelector  `json:"clusterSelector"`
+}
+
+type ManagedClusterSelector struct{
+    // "" means to use the current mechanism of matching label <cluster.open-cluster-management.io/clusterset:<ManagedClusterSet Name>.
+    // "LabelSelector" means to use the LabelSelector to select target managedClusters
+    // "ClusterNames" means to use "ClusterNames" where the managedClusters Name is set
+    // +optional
+    SelectorType SelectorType `json:"selectorType"`
+
+    LabelSelector *metav1.LabelSelector `json:"labelSelector"`
```

Comment:
in the godoc here, please indicate that `cluster.open-cluster-management.io/*` have secondary ACL checks and that other labels do *not* have any secondary ACL protection.

Reply:
done